trlx/pipeline/offline_pipeline.py

```diff
@@ -17,16 +17,25 @@ class PromptPipeline(BasePipeline):
 
     def __init__(self, prompts: List[str], max_prompt_length: int, tokenizer=None):
         super().__init__()
-
-        if tokenizer:
+        if tokenizer and 't5' in tokenizer.name_or_path:
+            model_inputs = tokenizer(prompts, truncation=True, padding="max_length", max_length=max_prompt_length)
```

Comment:
Is there any reason to truncate earlier tokens:
`<EARLY tokens> ... <RECENT tokens> ➡️ {EARLY tokens}`
instead of:
`<EARLY tokens> ... <RECENT tokens> ➡️ {RECENT tokens}`
Aren't we also `left` padding which suggests we can end up with more pad tokens than necessary this way?

I believe you mentioned this issue in the Discord but I'm not sure why `seq2seq` architectures would require a different truncation here 😅 Could you explain for reference? Thanks!

Reply:
Here are several points:
1. In the summary setup for the TL;DR dataset, the prompt input format of the dataset to train SFT is like this: 
```SUBREDDIT: r/relationships TITLE: Me [25F] with my friend [27F], she hasn't made .....```
After that when I do PPO by `trlx`, the prompt changed because in long text cases, keywords like: "SUBREDDIT", and "TITLE" ... could be removed, and with my experiments, it affected the result.
2. When you use `flant5` sometime you will add the instruction at the beginning like:
- Summarize this document: (CNN) The football match..... ---> same as the case above "Summarize this document" instruction could be removed.  
3. By default some tokenizers default truncated by `right` but in our implementation, we do truncate by `left` this may not be clear to users. As I know `RL4LMs` also have options in the configure file to choose `padding_side` and `truncation_side`, users can choose to fit with their setup.

Reply:
You're right Duy, it's not specific to `seq2seq` but rather to the task and I was about to suggest moving it as a config option as in `RL4LM`

Reply:
Thank you. I will try to add it to the config for this PR. :D